lib/services/database_test.go

```diff
@@ -384,46 +385,58 @@ func TestRDSTagsToLabels(t *testing.T) {
 
 // TestDatabaseFromRedshiftCluster tests converting an Redshift cluster to a database resource.
 func TestDatabaseFromRedshiftCluster(t *testing.T) {
-	cluster := &redshift.Cluster{
-		ClusterIdentifier:   aws.String("mycluster"),
-		ClusterNamespaceArn: aws.String("arn:aws:redshift:us-east-1:1234567890:namespace:u-u-i-d"),
-		Endpoint: &redshift.Endpoint{
-			Address: aws.String("localhost"),
-			Port:    aws.Int64(5439),
-		},
-		Tags: []*redshift.Tag{
-			{
-				Key:   aws.String("key"),
-				Value: aws.String("val"),
+	t.Run("success", func(t *testing.T) {
+		cluster := &redshift.Cluster{
+			ClusterIdentifier:   aws.String("mycluster"),
+			ClusterNamespaceArn: aws.String("arn:aws:redshift:us-east-1:1234567890:namespace:u-u-i-d"),
+			Endpoint: &redshift.Endpoint{
+				Address: aws.String("localhost"),
+				Port:    aws.Int64(5439),
 			},
-			{
-				Key:   aws.String("elasticbeanstalk:environment-id"),
-				Value: aws.String("id"),
+			Tags: []*redshift.Tag{
+				{
+					Key:   aws.String("key"),
+					Value: aws.String("val"),
+				},
+				{
+					Key:   aws.String("elasticbeanstalk:environment-id"),
+					Value: aws.String("id"),
+				},
 			},
-		},
-	}
-	expected, err := types.NewDatabaseV3(types.Metadata{
-		Name:        "mycluster",
-		Description: "Redshift cluster in us-east-1",
-		Labels: map[string]string{
-			types.OriginLabel: types.OriginCloud,
-			labelAccountID:    "1234567890",
-			labelRegion:       "us-east-1",
-			"key":             "val",
-		},
-	}, types.DatabaseSpecV3{
-		Protocol: defaults.ProtocolPostgres,
-		URI:      "localhost:5439",
-		AWS: types.AWS{
-			AccountID: "1234567890",
-			Region:    "us-east-1",
-			Redshift: types.Redshift{
-				ClusterID: "mycluster",
+		}
+		expected, err := types.NewDatabaseV3(types.Metadata{
+			Name:        "mycluster",
+			Description: "Redshift cluster in us-east-1",
+			Labels: map[string]string{
+				types.OriginLabel: types.OriginCloud,
+				labelAccountID:    "1234567890",
+				labelRegion:       "us-east-1",
+				"key":             "val",
 			},
-		},
+		}, types.DatabaseSpecV3{
+			Protocol: defaults.ProtocolPostgres,
+			URI:      "localhost:5439",
+			AWS: types.AWS{
+				AccountID: "1234567890",
+				Region:    "us-east-1",
+				Redshift: types.Redshift{
+					ClusterID: "mycluster",
+				},
+			},
+		})
+
+		require.NoError(t, err)
+
+		actual, err := NewDatabaseFromRedshiftCluster(cluster)
+		require.NoError(t, err)
+		require.Equal(t, expected, actual)
+	})
+
+	t.Run("missing endpoint", func(t *testing.T) {
+		_, err := NewDatabaseFromRedshiftCluster(&redshift.Cluster{
+			ClusterIdentifier: aws.String("still-creating"),
+		})
+		require.Error(t, err)
+		require.True(t, trace.IsBadParameter(err), fmt.Sprintf("expected trace.BadParameter, got %v", err))
```

Comment:
Drop the printf, require functions do formatting:

```suggestion
		require.True(t, trace.IsBadParameter(err), "Expected trace.BadParameter, got %v", err)
```

I would drop the check above too, as this ends up doing both and has an error description. Up to you.

Reply:
ah, nice. removed the fmt. i think i will keep the `require.Error`.